test/common/network/io_socket_handle_impl_test.cc

```diff
@@ -79,25 +82,34 @@ TEST(IoSocketHandleImpl, LastRoundTripTimeReturnsRttIfSuccessful) {
   auto os_calls =
       std::make_unique<Envoy::TestThreadsafeSingletonInjector<Envoy::Api::OsSysCallsImpl>>(
           &os_sys_calls);
-  EXPECT_CALL(os_sys_calls, getsockopt_(_, _, _, _, _))
-      .WillOnce(DoAll(WithArg<3>(Invoke([](void* optval) {
-                        static_cast<struct tcp_info*>(optval)->tcpi_rtt = 35;
-                      })),
-                      Return(0)));
+
+  EXPECT_CALL(os_sys_calls, socketTcpInfo(_, _))
+      .WillOnce(Invoke([this](os_fd_t sockfd, tcp_info* tcpInfo) -> Api::SysCallBoolResult {
+        tcpInfo->tcpi_rtt = 35;
+        return {true, 0};
+      }));
 
   IoSocketHandleImpl io_handle;
   EXPECT_THAT(io_handle.lastRoundTripTime(), Eq(absl::optional<std::chrono::milliseconds>{35}));
 }
 
-#endif
-
-#ifndef TCP_INFO
-
-TEST(IoSocketHandleImpl, LastRoundTripTimeAlwaysReturnsEmptyOptional) {
-  IoSocketHandleImpl io_handle;
-  EXPECT_THAT(io_handle.lastRoundTripTime(), Eq(absl::optional<std::chrono::milliseconds>{}));
+// Only do the integration tests in supported platforms.
+#if defined(TCP_INFO) || defined(SIO_TCP_INFO)
+TEST(IoSocketHandleImpl, LastRoundTripIntegrationTest) {
+  struct sockaddr_in server;
+  // TCP info can not be calculated on loopback.
+  // For that reason we a public dns server.
+  server.sin_addr.s_addr = inet_addr("1.1.1.1");
```

Comment:
Is there precedence in the Envoy tests for connecting to remote endpoints? This adds an internet connectivity dependency, which could be flaky.

Reply:
Fair point and open for further discussion. 

The reasons I went with a remote endpoint:

1. If we only mock this functionality given that this code is optional it becomes a big blind spot. If a regression is introduced in this call it will be extremely hard to detect.
2. I assume public dns servers (either from cloudflare or google) are highly available.
3. It is easier to be optimistic and disable the test if it turns out to reduce developer velocity than to add it afterwards.
4. There is no good alternative to test on localhost

Reply:
Is there a way to somehow annotate this test as optional or manually run only? Based on the slack discussion it looked like Yan also had some reservations. I would agree that its nice to have the test in place anytime someone needs to work in the space and verify their changes, but a flaky automated test will also lower productivity and cause frustration. FWIW I tried searching the test code and couldn't really find a concrete example of connecting to remote endpoints

Reply:
@envoyproxy/senior-maintainers any thoughts/preference on this?

Reply:
maybe tag this as a `manual` test?

Reply:
fixed in 7c9137b7a3fb86ccae63cbd4018683cc47b7cb4d

Reply:
Don't tag it `manual`, that prevent this test from built, just tag this `requires-network`, as documented in https://docs.bazel.build/versions/master/be/common-definitions.html